--- rubygem-passenger.te.orig	2011-09-04 15:00:58.000000000 -0700
+++ rubygem-passenger.te	2011-09-04 15:01:26.000000000 -0700
@@ -66,7 +66,7 @@
 # Nor var_run files & sockets
 allow httpd_passenger_helper_t httpd_t:file getattr;
 allow httpd_passenger_helper_t httpd_var_run_t:dir { write remove_name search getattr add_name };
-allow httpd_passenger_helper_t httpd_var_run_t:file { write create open setattr };
+allow httpd_passenger_helper_t httpd_var_run_t:file { write create setattr };
 allow httpd_passenger_helper_t httpd_var_run_t:sock_file { write create unlink setattr };
 
 allow httpd_passenger_helper_t self:capability sys_ptrace;
