diff --git a/lib/phusion_passenger/standalone/runtime_installer.rb b/lib/phusion_passenger/standalone/runtime_installer.rb
index 9ac15c2..d732e72 100644
--- a/lib/phusion_passenger/standalone/runtime_installer.rb
+++ b/lib/phusion_passenger/standalone/runtime_installer.rb
@@ -217,6 +217,7 @@ private
 		end
 		total_progress = (phase.first - 1).to_f / total_phases
 		total_progress += (progress.to_f / total) * ((phase.last - phase.first + 1).to_f / total_phases)
+		total_progress = 0 if  total_progress < 0
 		
 		max_width = 79
 		progress_bar_width = 45
